src/components/RunGraph.vue

```diff
@@ -49,7 +51,11 @@
 
   watch(() => props.selected, selected => selectNode(selected))
 
-  emitter.on('nodeSelected', nodeId => emit('update:selected', nodeId))
+  emitter.on('nodeSelected', selection => emit('update:selected', selection))
+
+  emitter.on('inspectArtifact', artifact => {
+    emit('inspectArtifact', artifact)
+  })
```

Comment:
it be nice to get some consistent naming and terminology going here. We already have `NodeSelection` as a possible value for `selected`. Could we add an `ArtifactSelection` type to that and reuse the same event?

Reply:
Yeah so, just trying to think ahead for when we have a series of things that emit a value for showing pop overs (or a drawer in the case of artifacts), it seems like these should be distinct from selections which have a relationship with the selectedNode state. I could make this `inspect` emitter more generic, then include a `type` so the UI knows what popup and content to show. Does it make sense to manage `type` rather than just having separate events? Then the data passed in the event can be what's best for the type. Like I think I'll actually end up including the artifact `key` (name) as well so that a link can be seen quickly in the flyout while the rest of the (possibly big) artifact loads. Thoughts on that? The picture could get clearer as those other layers are added on too.

Reply:
You've got me thinking artifacts should appear selected when clicked now.. Possibly other "inspectable" things as well. Probably a cleaner paradigm for sure. Will mess with that.

Reply:
Yeah it get's all mixed up with the highlighting logic, I'm still thinking they need to be separated.

Reply:
what do you mean gets mixed up? Because a node needs to be highlighted at the same time as a artifact? 

I'm thinking selected would be a discriminating union so no matter what is selected we'd know what type of thing it is and any necessary data about it. You can only select/click on one thing at a time right? So having multiple events for selection doesn't seem like a good pattern to me.

Reply:
Nodes subscribe to selection updates and dim themselves if they're not selected. I suppose the consumer can set the selection to null if it's not a node, but that would prevent showing any sort of selected state on the artifact so I'm not sure what the point would be then. The interactions differ in that nodes are selectable, artifacts, events, and automations are "inspectable", so you click them to trigger a popover, but you're not actually selecting them the same way as you would a node, where the selection is persisted in state and updates the view. I do still totally think you're on to something though. I'll throw something on the calendar for a quick sync discussion on it.